### ISSUE TITLE
Doesn't open on ubuntu 18.10

### ISSUE THREAD

**Issue by nklayman** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3391]
**Checklist:**

* [x] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES ASWELL**
* [x] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [x] I have double checked and can reproduce the issue

**OS:** Ubuntu 18.10

**GPMDP Version:** 4.6.1
 
**Issue Descriptions:**
App doesn't open whatsoever. When I run via command line, I get:
```
[1]    9872 segmentation fault (core dumped)  google-play-music-desktop-player

~  
```
It worked fine on ubuntu 18.04, but not on 18.10 (tried on two different computers, sent a crash report from one).
**Steps to Reproduce:**
Install on ubuntu 18.10
Try to open it.

Thanks for your help!

P.S. It seems similar to #3117, but libgconf-2-4 was already installed.

**Comment by manicholls** [comment]
Seeing the same.  libgconf-2-4 is installed and this is a fresh install of 18.10.

**Comment by Steampunkery** [comment]
I have the same problem. Attempting to build from source right now.

**Comment by nklayman** [comment]
I got that error when I tried to build from source as well.

**Comment by zerkz** [comment]
It failed for me too, I'll see if I can fix the build from source sometime.

**Comment by zerkz** [comment]
@jostrander After getting my dev dependencies in order on 18.10, I can still confirm it crashes (seems to be a SEGFAULT?) even after building straight from source.

**Comment by Steampunkery** [comment]
Same here. Just finished my build. I would have replied earlier but I uninstalled node after it failed to build yesterday because I had only installed it for that purpose 😆

**Comment by jostrander** [comment]
Possibly something to do with this? https://github.com/electron/electron/issues/13972

**Comment by zerkz** [comment]
@jostrander i just found that and was about to post this. I believe this is definitely related.

**Comment by JoshuaGarrison27** [comment]
@zerkz I'm not sure why was I tagged on this. I am only running on Windows.

**Comment by zerkz** [comment]
@JoshuaGarrison27 doh, my bad, meant to tag @jostrander :joy:

**Comment by jostrander** [comment]
That might be the case @zerkz but I can't test it unfortunately. I do know that the 3.x release of electron works out of the box as I've been testing it locally on my machine lately, so you shouldn't have any other issues upgrading it.

**Comment by zerkz** [comment]
@jostrander I'm building with 2.0.8 as we speak.

**Comment by zerkz** [comment]
@jostrander Bam! it worked. Opening a PR right now.

**Comment by jostrander** [comment]
Awesome! Glad to hear, I have more changes coming soon that will bump it to the 3.x branch so it may not be necessary.

**Comment by zerkz** [comment]
https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/pull/3400

**Comment by christofdamian** [comment]
FYI the same problem happens on Fedora 29 which will be released today.

**Comment by GCorbel** [comment]
The version with the fix is not released yet. If you want to download a working version, you have to go to : https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL- and the check the last build for "#master". Then, go to "artifacts" and download the package you want.

This is for deb 64 bits : 

https://2838-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb

And this for deb 32 bits : 

https://2838-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_i386.deb

**Comment by agc93** [comment]
Okay, maybe I'm going mad, but I cannot see an Artifacts tab for any recent build in CircleCI.

For any Fedora users who end up here, your latest URLs (as of 19/12/18) are:

[https://2911-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm](https://2911-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm)

[https://2911-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.i386.rpm](https://2911-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.i386.rpm)

**Comment by nklayman** [comment]
Make sure the build varient is artifactcollector instead of Debian or fedora.

**Comment by agc93** [comment]
@nklayman Even in the artifact_gather jobs, I'm still not seeing anything?

![screenshot from 2018-12-21 10-39-21](https://user-images.githubusercontent.com/1369269/50318434-d7379000-050c-11e9-950f-50bc04566789.png)

**Comment by nklayman** [comment]
I think you need to sign in to circleci to see artifacts. If you can't, I can send you a link to the latest one. What platform are you on?

**Comment by agc93** [comment]
@nklayman I'm on Fedora, but I've already got the links (see comments above :smile: ). I don't have a CircleCI account, but I've got the link pattern sorted so will have to use that, thanks.

**Comment by jwulf** [comment]
WORKAROUND: If you install Visual Studio Code, you can use the working libnode.so from that to get this to work on Fedora 29.

Just run this command: `sudo ln -fs /usr/share/code/libnode.so /usr/share/google-play-music-desktop-player/libnode.so`

This links the working libnode from VS Code to the player.

**Comment by zerkz** [comment]
@MarshallOfSound Any plans for a release soon? seems folks are still hitting this issue a lot.

**Comment by silentdis** [comment]
It's good to hear there's some fix for this... but after trying to figure out how to build from the repo for an hour and failing on various dependencies, I've given up again.

Back to 4.4.x with me once again.

**Comment by knuckleheadTech** [comment]
Little confused how this is closed. on Fedora 29 issue still persists. Went as far as to try the flatpak as well but it has the same issue (realize not related here but just for reference).

Have no need for Visual Studio and did the work around using Atom. But a workaround isn't really a fix. It is likely I have missed something obvious.

`sudo dnf install ~/Downloads/atom.x86_64.rpm
sudo mv /usr/lib/slack/libnode.so /usr/lib/slack/libnode.so.bad
sudo ln -s /usr/share/atom/libnode.so /usr/lib/slack/libnode.so`

[Second Answer](https://stackoverflow.com/questions/53084955/why-does-slack-return-a-segmentation-fault-after-fedora-29-upgrade)

**Comment by jostrander** [comment]
I understand the confusion. It's closed because it's technically fixed in master, but it hasn't been released.

**Comment by knuckleheadTech** [comment]
Gotcha. Went to grab latest master but I guess with no account I don't see it. 

Thought I was clever with the libnode fix but now it loads to the old blackscreen issue that from was a few releases ago. Pretty sure this is a dependency issue but it's been long enough ago I forget. Would you mind linking the rpm from the latest fixed master?

**Comment by jostrander** [comment]
https://2911-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm

**Comment by pb1051** [comment]
> https://2911-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm

This is working for me on Fedora 29

**Comment by GCorbel** [comment]
I did a gif to explain what to do : 

![peek 15-02-2019 11-57](https://user-images.githubusercontent.com/1296201/52871880-10f5d300-3119-11e9-95d2-9533fbf6c2a2.gif)

You have to be logged to see the "artifact tab" and go to https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/tree/master .

**Comment by ringerc** [comment]
for Google-ability:

If you're seeing a backtrace like

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000000a4fa30 in ?? ()
Missing separate debuginfos, use: dnf debuginfo-install google-play-music-desktop-player-4.6.1-1.x86_64
(gdb) bt
#0  0x0000000000a4fa30 in  ()
#1  0x00007f60179bd2b2 in node::http2::Http2Session::Callbacks::Callbacks(bool) () at /usr/share/google-play-music-desktop-player/libnode.so
#2  0x00007f60179bd375 in  () at /usr/share/google-play-music-desktop-player/libnode.so
#3  0x00007f6017c17e0a in call_init.part () at /lib64/ld-linux-x86-64.so.2
#4  0x00007f6017c17f0a in _dl_init () at /lib64/ld-linux-x86-64.so.2
#5  0x00007f6017c0914a in _dl_start_user () at /lib64/ld-linux-x86-64.so.2
#6  0x0000000000000001 in  ()
#7  0x00007fff0448b7bd in  ()
#8  0x0000000000000000 in  ()
(gdb) q
```

then this is the issue you're hitting.

I suggest re-titling the issue since this appears to affect x64 Linux in general.

**Comment by marquesarthur** [comment]
Adding a comment for Fedora 29:

Make sure you don't have any garbage from a failed installation:
```
sudo rpm -qa | grep google-play-music-desktop-player 
sudo yum remove google-play-music-desktop-player 
```
Download the package mentioned by @pb1051 and @jostrander 
Finally, install it with:

```yum install google-play-music-desktop-player-4.6.1.x86_64.rpm```

It will install all the necessary dependencies and or boi runs nice and smoothly afterwards.
Thanks, guys

**Comment by webmasterMeyers** [comment]
the quickest fix for me on Ubuntu 18.10 was `sudo apt-get purge google-play-music-desktop-player` to remove the broken install.  Then use snap, `sudo snap install google-play-music-desktop-player`

**Comment by Kernald** [comment]
Any chance on seeing a 4.6.2 or 4.7.0 or whatever actually being released? It's been a year since this issue has been noticed in 4.6.1, and almost a year since it has been fixed, and yet workarounds are still needed...

**Comment by kavika13** [comment]
I did a wipe of my install (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3391#issuecomment-474522652), and installed the RC version I found here: https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3548 - this fixed the problem for me on Pop!_OS (which is basically Ubuntu/Debian).

**Comment by Kernald** [comment]
> I did a wipe of my existing 4.6.1-1 install ([#3391 (comment)](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3391#issuecomment-474522652)), and installed the RC version I found here: #3548 - this fixed the problem for me on Pop!_OS (which is basically Ubuntu/Debian).

The problem is, the RC you linked to is just that - a release candidate. Any distribution packaging GPMDP is gonna have the previously released 4.6.1 - which, as this issue shows, doesn't work anymore.

**Comment by kavika13** [comment]
> The problem is, the RC you linked to is just that - a release candidate. Any distribution packaging GPMDP is gonna have the previously released 4.6.1 - which, as this issue shows, doesn't work anymore.

Yes, I agree with you. I commented because I found a work-around that met my personal criteria, and didn't see it mentioned on the thread.

TBH a showstopper issue with a working fix should have warranted a hotfix release a year ago, or whenever it got fixed. The maintainer should probably solicit help if they can't manage to do that sort of thing in a timely manner.
